hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/states/endpoint/VersionEndpointTask.java

```diff
@@ -106,7 +106,8 @@ public VersionEndpointTask(EndpointStateMachine rpcEndPoint,
           volumeSet.writeUnlock();
         }
 
-        ozoneContainer.getDispatcher().setScmId(scmId);
+        // Start the container services after getting the version information
+        ozoneContainer.start(scmId);
```

Comment:
How does the fix work? I don't understand this startup sequence well enough to figure it out.

Reply:
The issue is arising because after a restart, DN is accepting client requests before registering with SCM. We were starting the container services (read & write channels) before getting the version information. I have moved the services start command to the end of getVersionInfo task.